ckanext/ontario_theme/templates/internal/package/resource_validation.html

```diff
@@ -15,27 +15,80 @@
     {% set success_message = "Data validated successfully" %}
     <div class="validation-details">
       <div>{{ _('Validation timestamp') }}: {{ h.render_datetime(resource.validation_timestamp, with_hours=True) }}</div>
       <div>{{ _('Duration') }}: {{ h.validation_dict(validation.report)["stats"]["seconds"] }}s</div>
     </div>
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
+
+    <script>
+      function poll_datastore() {
+        const resId = "{{ resource.id }}";
+        const xhr = new XMLHttpRequest();
+        const url = `/api/3/action/resource_show?id=${resId}`
+
+        xhr.open("POST", url, true);
+        xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+        xhr.send();
+
+        xhr.onreadystatechange = () => {                                
+          if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+            let thisResponse = JSON.parse(xhr.response);
+
+            if (thisResponse.result.datastore_active) {
+              const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
+         
+              const fixedDelay = async () => {
+                await sleep(2000)
+                document.getElementById("pendingTxt").innerHTML = "Data file uploaded. Please click Continue.";
+                document.getElementById("spinnerSVG").style.display = "none";
+                document.getElementById("step2Btn").style.display = "block";
+              }
+              fixedDelay();
+
+            } else {
+            setTimeout(poll_datastore, 1000);
+            }
+          }
+        }
+      };
+      poll_datastore();
+    </script>
+
+    <div class="ontario-small-12 ontario-row">
+      <div class="ontario-small-12 ontario-columns">
+        <div id="loadingSpinner" aria-hidden="false" role="alert">
+          <div class="ontario-loading-indicator">
+            <svg id="spinnerSVG"
+                 class="ontario-loading-indicator__spinner-small"
+                 viewBox="25 25 50 50"
+                 xmlns="http://www.w3.org/2000/svg">
+              <circle cx="50" cy="50" r="20" fill="none" stroke-width="4" />
+            </svg>
+            <p class="spinner-small" id="pendingTxt">Uploading data file to database</p>
+          </div>
+        </div>
+      </div>
+    </div>
+
   {% else %}
+    {% set button_id = "nextBtn" %}
     {% set success_message = "Data uploaded successfully" %}
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
   {% endif %}
 
-  {% set package_id = resource.get('package_id') %}
-  {% set resource_id = resource.get('id') %}
-  {% set action = h.url_for('datastore.dictionary',
-      id=package_id,
-      resource_id=resource.id) %}
- 
-  <form method="get" action="{{ action }}">
-    <button class="ontario-button ontario-button--primary"
+  <form "method=post" action="{{ action }}">
```

Comment:
Why is method=post invalid HTML? Can you please add a reference.

Reply:
Core CKAN is using method=post and not get.

Reply:
`method="post"` is valid `"method=post"` is not

Reply:
"method=post" is your code

Reply:
What do you mean it is my code? In this [commit](https://github.com/ongov/ckanext-ontario_theme/pull/585/commits/4532524ce5b87ba656dfbefc497a414b3e1c927d) the form was changed from a form with `method="get"` to an anchor tag and then in this [commit](https://github.com/ongov/ckanext-ontario_theme/pull/585/commits/8596d02413d969e7b6386afa8d878261c11067cd) it was changed from an anchor tag to a form with `"method=post"`, they were both committed by you.

Reply:
https://github.com/ongov/ckanext-ontario_theme/blob/ckanext_validation_dev_DATA1439_poll/ckanext/ontario_theme/templates/internal/datastore/dictionary.html#L24

Reply:
That would be added as a string which the browser would interpret as `method="post"` screenshot:
![image](https://github.com/user-attachments/assets/edfd70bf-2517-4a4f-9263-dcd92960f537)
With `<form "method=post"...` it is invalid to the browser screenshot:
![image](https://github.com/user-attachments/assets/0bd70a5f-47e1-4a97-8564-29e08fd42eb0)

Reply:
So your issue is with the quotes and not the method.
It needs to be post because it gets processed in `def post` here: https://github.com/ongov/ckanext-ontario_theme/blob/ckanext_validation_dev_DATA1439_poll/ckanext/ontario_theme/datastore.py#L63

Reply:
The form on the data dictionary page uses POST so that is what is calling the datastore function `def post`. Since the browser was not interpreting the `"method="post""` correctly, the form was defaulting to GET. The data dictionary page worked fine. If you were to try to use `method="post"` you would get an internal server error.

Reply:
Resolved! To recap:
- method=post is not invalid html per se, it was the quotes
- for this button the method should be `get` because it directs to the `def get` method of the dictionary page: https://github.com/ongov/ckanext-ontario_theme/blob/ckanext_validation_develop/ckanext/ontario_theme/datastore.py#L52 (and not the post method, my bad)